bin/webpack-dev-server.js

```diff
@@ -330,14 +330,14 @@ function startDevServer(wpOpt, options) {
 	const protocol = options.https ? "https" : "http";
 
 	// the formatted domain (url without path) of the webpack server
-	const domain = url.format({
+	const domain = options.public ? `${protocol}://${options.public}` : url.format({
 		protocol: protocol,
 		hostname: options.host,
 		port: options.socket ? 0 : options.port.toString()
 	});
 
 	if(options.inline !== false) {
-		const devClient = [`${require.resolve("../client/")}?${options.public ? `${protocol}://${options.public}` : domain}`];
```

Comment:
this line behaves the same, but domain is also used on line 375 to calculate the url that gets opened, and that's where the behavior changes